public/app/plugins/panel/stat/common.test.ts

```diff
@@ -0,0 +1,113 @@
+import { FieldType, FieldDisplay, ArrayVector } from '@grafana/data';
+
+import { formatDisplayValuesWithCustomUnits } from './common';
+
+describe('formatting functions', () => {
+  describe('formatDisplayValuesWithCustomUnits', () => {
+    const prefix = '&';
+    const suffix = '*';
+    const emptyArryVector = new ArrayVector([]);
+    const mockSharedFieldValueProps = {
+      hasLinks: false,
+      display: {
+        color: '#F2495C',
+        numeric: 95.13749301705772,
+        percent: 0.5197488596479736,
+        prefix: '$',
+        suffix: '%',
+        text: '95.1',
+        title: 'A-series',
+      },
+    };
+    const mockFieldValues: FieldDisplay[] = [
+      {
+        name: 'test1',
+        field: {},
+        sourceField: { type: FieldType.number, name: 'test1', config: {}, values: emptyArryVector },
+        ...mockSharedFieldValueProps,
+      },
+      {
+        name: 'test2',
+        field: {},
+        sourceField: { type: FieldType.time, name: 'test2', config: {}, values: emptyArryVector },
+        ...mockSharedFieldValueProps,
+      },
+      {
+        name: 'test3',
+        field: {},
+        sourceField: { type: FieldType.boolean, name: 'test13', config: {}, values: emptyArryVector },
+        ...mockSharedFieldValueProps,
+      },
+    ];
+
+    it('passes the untouched field values through if no custom prefixes are present', () => {
+      const unchangedFieldValues = formatDisplayValuesWithCustomUnits(mockFieldValues, {
+        // Empty custom values
+        prefix: '',
+        suffix: '',
+      });
+
+      // toEqual() recursively checks every field of an object or array for equality
+      expect(unchangedFieldValues).toEqual(mockFieldValues);
+    });
+    it('sucessfully prepends the `display.prefix` value with the custom prefix, and leave the suffix unchanged', () => {
```

Comment:
Super minot nit - I think having a line break between tests improves readability
```suggestion
    });

    it('sucessfully prepends the `display.prefix` value with the custom prefix, and leave the suffix unchanged', () => {
```

Reply:
Agreed. On it.